### ISSUE TITLE
java.lang.InstantiationError when using diffx with java annotations

### ISSUE THREAD

**Issue by vitaliihonta** [softwaremill/diffx#45]
Hi)

I've found a tricky error when matching over case class annotated with java annotations.
Minimal example:
```scala
// in src/main dir:
package com.example

@Deprecated
case class Bar(x: String, y: Int)

// in src/test dir
package com.example

import com.softwaremill.diffx.scalatest.DiffMatcher
import org.scalatest.{FlatSpec, Matchers}

class TestSpec extends FlatSpec with Matchers with DiffMatcher {
  "diffx" should "work" in {
    Bar(x = "a", y = 1) should matchTo(
      Bar(x = "a", y = 0)
    )
  }
}

```

Then I've got the following exception:
```
[info] com.example.TestSpec *** ABORTED ***
[info]   java.lang.InstantiationError: java.lang.Deprecated
[info]   at com.example.TestSpec$$anon$1.<init>(TestSpec.scala:8)
[info]   at com.example.TestSpec.$anonfun$new$1(TestSpec.scala:8)
// and then some scalatest/sbt stack trace mase
```

Removing annotation solves this exception, but in some cases it's not an option.
(for instance when using swagger annotations).

Is it a known behavior?

**Comment by vitaliihonta** [comment]
My sbt prject setup:
```scala
name := "Tmp"
version := "0.1"
scalaVersion := "2.12.8"

libraryDependencies ++= {
  Seq(
    "org.scalatest" %% "scalatest" % "3.0.8",
    "com.softwaremill.diffx" %% "diffx-scalatest" % "0.3.10"
  )
}
```

**Comment by ghostbuster91** [comment]
Thanks for reporting. This is really weird. I've checked and it seems to be a bug in magnolia, see https://github.com/propensive/magnolia/issues/187.

**Comment by vitaliihonta** [comment]
Thank you! I'll be watching magnolia's issue

**Comment by ghostbuster91** [comment]
Fixed in 0.3.12
